Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  Alicerce.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public final class Log {
+
+    private static var providers = [LogProvider]()
+
+    public static let defaultLevel: Level = Level.error
+
+    // MARK:- Provider Management
+
+    internal static var providerCount: Int {
+        return providers.count
+    }
+
+    public class func register(_ provider: LogProvider) {
```

Comment:
I personally would prefer if we had a protocol called `LogStack` or something, which would contain the logging API's (error, warning, debug, etc) and then a `MultiDestinationLogStack`/`MultiLogStack`/`Log` class that would conform to that protocol. 

This class would then be used as a service that is injected into every object that requires logging, without even needing to be a singleton (we could still have a `.default`/`sharedInstance`singleton property for people that still like to use singletons 😓).

This would enable a `DummyLogStack` which would do nothing, a simple `ConsoleLogStack`, etc. 

I would also prefer if the API's weren't `static`, but using a protocol it isn't so "problematic".

Using this approach makes the Logging infrastructure more flexible, allows easier testing by injecting a custom Logger, makes creating simpler loggers easier and decouples the user from having to use the same logging instance.

What do you say?